Core/include/Acts/Propagator/MaterialInteractor.hpp

```diff
@@ -55,74 +56,85 @@ struct MaterialInteractor {
   void operator()(propagator_state_t& state, const stepper_t& stepper,
                   const navigator_t& navigator, result_type& result,
                   const Logger& logger) const {
-    // In case of Volume material update the result of the previous step
-    if (recordInteractions && !result.materialInteractions.empty() &&
-        !result.materialInteractions.back().volume.empty() &&
-        result.materialInteractions.back().updatedVolumeStep == false) {
-      updateResult(state, stepper, result);
-    }
-
-    // If we are on target, everything should have been done
-    if (navigator.targetReached(state.navigation)) {
+    if (state.stage == PropagatorStage::postPropagation) {
       return;
     }
+
     // Do nothing if nothing is what is requested.
     if (!(multipleScattering || energyLoss || recordInteractions)) {
       return;
     }
-    // We only have material interactions if there is potential material
-    const Surface* surface = navigator.currentSurface(state.navigation);
-    const TrackingVolume* volume = navigator.currentVolume(state.navigation);
 
-    if (!(surface && surface->surfaceMaterial()) &&
-        !(volume && volume->volumeMaterial())) {
-      return;
-    }
+    // Handle surface material
 
+    // Note that start and target surface conditions are handled in the
+    // interaction code
+    const Surface* surface = navigator.currentSurface(state.navigation);
+
+    // We only have material interactions if there is potential material
     if (surface && surface->surfaceMaterial()) {
+      ACTS_VERBOSE("MaterialInteractor | "
+                   << "Found material on surface " << surface->geometryId());
+
       // Prepare relevant input particle properties
       detail::PointwiseMaterialInteraction d(surface, state, stepper);
```

Comment:
Not part of the change set, but `d` really is that great of a variable name for this.